frontend/lib/src/theme/utils.ts

```diff
@@ -392,6 +390,11 @@ export const setCachedTheme = (themeConfig: ThemeConfig): void => {
 
   deleteOldCachedThemes()
 
+  // Do not set the theme if the app has a pre-defined theme from the embedder
+  if (isLightTheme() || isDarkTheme()) {
```

Comment:
nit: The `isDarkTheme` and `isLightTheme` namings are slightly confusing. Maybe rename it to something more explicit, e.g., `isLightThemeInQueryParams`.